src/components/Card.js

```diff
@@ -1,41 +1,69 @@
+import { data } from "autoprefixer";
+
 export default class Card {
-  constructor(data, cardSelector, handleImageClick){
-    this.title = data.title;
-    this.url = data.url;
+  constructor(data, cardSelector, handleImageClick, handleDeleteClick, handleLikeClick, handleDislikeClick){
+    this.name = data.name;
+    this.link = data.link;
+    this._id = data._id;
+    this._isLiked = data.isLiked
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._handleDeleteClick = handleDeleteClick
+    this._handleLikeClick = handleLikeClick;
+    this._handleDislikeClick = handleDislikeClick;
+    this._cardElement = document.querySelector(this._cardSelector).content.querySelector('.card').cloneNode(true);
+    this._cardImageElement = this._cardElement.querySelector('.card__image');
   };
   
-  _delete(){
-    this._cardElement.remove();
+  getCardId(){
+    return this._id
+  }
+
+  delete(){
+    this._cardElement.remove()
   };
   
-  _like(){
-    this._cardElement.querySelector('.card__heart').classList.toggle("card__heart_active");
+  like(){
+    this._cardElement.querySelector('.card__heart').classList.add("card__heart_active");
+    this._isLiked = !this._isLiked
   };
 
-  _setEventListeners(){
-    this._cardImageElement = this._cardElement.querySelector('.card__image');
+  dislike(){
+    this._cardElement.querySelector('.card__heart').classList.remove('card__heart_active');
+    this._isLiked = !this._isLiked
+  }
 
+  _setEventListeners(){
     this._cardElement.querySelector('.card__heart').addEventListener('click', ()=>{
-      this._like();
+      if (this._isLiked){
+        this._handleDislikeClick(this.getCardId())
+      } else {
+        this._handleLikeClick(this.getCardId())
+      }
+      
     });
 
     this._cardElement.querySelector('.card__delete').addEventListener('click', ()=>{
-      this._delete();
+      this._handleDeleteClick(this)
     });
 
     this._cardImageElement.addEventListener('click', ()=>{
       this._handleImageClick(this);
     });
-    };
+  };
+
+  renderLike(){
+    this._cardElement.querySelector('.card__heart').classList.add('card__heart_active')
+  }
 
-  
   generateCard(){
-    this._cardElement = document.querySelector(this._cardSelector).content.querySelector('.card').cloneNode(true);
-    this._cardElement.querySelector('.card__image').src = `${this.url}`;
-    this._cardElement.querySelector('.card__image').alt = `${this.title}`;
-    this._cardElement.querySelector('.card__title').textContent = `${this.title}`;
+    this._cardElement.querySelector('.card__image').src = `${this.link}`;
```

Comment:
`this._cardElement.querySelector('.card__image')`

You've found the image as `this._cardImageElement` in lne 15


Please, use it here

`Class fields` (the so called **class variables** with `this`) are accessible inside any method of the class. You shouldn’t pass **class fields** into the call of any method or search them again because you can use class fields inside all methods.